File: math/13. Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        res = 0
        dict = {"I":1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000}
        
        for i in range(0, len(s)-1):
            if dict[s[i]] < dict[s[i+1]]:
                res -= dict[s[i]]
            else:
                res += dict[s[i]]
        
        res += dict[s[-1]]
        return res

File: math/13. Roman to Integer.c


int toNumber(char c){
    switch(c){
    case 'I':  
        return 1;  
    case 'V':  
        return 5;  
    case 'X':  
        return 10;  
    case 'L':  
        return 50;  
    case 'C':  
        return 100;  
    case 'D':  
        return 500;  
    case 'M':  
        return 1000;  
    default:
        return;
    }
}


int romanToInt(char* s) {
    int res = 0;
    int N = strlen(s);
    for(int i=0; i<N; i++){
        if(i!=N-1){
        if (toNumber(s[i]) < toNumber(s[i+1])){
            res = res - toNumber(s[i]);
        } else {
            res = res + toNumber(s[i]);
        }
    }else{res = res + toNumber(s[N-1]);}
    }
    return res;
}
